src/app.rs

```diff
@@ -67,26 +62,54 @@ impl App {
         let cargo_file_contents = format!(
             cargo_template_str!(),
             self.args.project_name(),
-            self.args.details());
+            self.args.details()
+        );
 
-        let main_file = DiskEntry::new(
-            MAIN_FILE_PATH,
-            MAIN_FILE_NAME,
-            DiskEntryType::File);
+        let main_file = DiskEntry::new(MAIN_FILE_PATH, MAIN_FILE_NAME, DiskEntryType::File);
 
-        let cargo_file = DiskEntry::new(
-                  CARGO_FILE_PATH,
-                  CARGO_FILE_NAME,
-            DiskEntryType::File);
+        let cargo_file = DiskEntry::new(CARGO_FILE_PATH, CARGO_FILE_NAME, DiskEntryType::File);
 
-        file_manager.write_to_file(
-            &main_file.get_full_path(),
-            main_template_str!().as_bytes())?;
+        file_manager.write_to_file(&main_file.get_full_path(), main_template_str!().as_bytes())?;
 
-        file_manager.write_to_file(
-            &cargo_file.get_full_path(),
-            cargo_file_contents.as_bytes())?;
+        file_manager.write_to_file(&cargo_file.get_full_path(), cargo_file_contents.as_bytes())?;
 
         Ok(base_path)
     }
 }
+
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempfile::tempdir;
+    use crate::args::AppArguments;
+
+    #[test]
+    fn test_app_run_success() {
+        let temp_dir = tempdir().unwrap();

```

Comment:
Cool, i think that's it!!
There's a pattern that i like to follow, it's called AAA Test pattern, where you split the initialization, the execution and the assertion into smaller 'sections'
[Here](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80) is a good article about it.
Once you start using it, you never stop, its a lot easier for understanding what the test does.
(Not a required change, just a suggestion)